scripts/mcp_control_plane_smoke.py

```
@@ -0,0 +1,253 @@
+
+
+def _require_tools(names: Sequence[str], required: Sequence[str]) -> None:
+    missing = [name for name in required if name not in names]
+    if missing:
+        raise RuntimeError(
+            "Penguin MCP server did not expose required tools: "
+            f"{missing}; available={names}"
+        )
+
+
+async def _run_smoke(server_command: Sequence[str], keep_workspace: bool) -> Dict[str, Any]:
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)